common/lib/dependabot/git_metadata_fetcher.rb

```diff
@@ -48,20 +48,20 @@
         # Remove the opening clause of the upload pack as this isn't always
         # followed by a line break. When it isn't (e.g., with Bitbucket) it
         # causes problems for our `sha_for_update_pack_line` logic
-        line = upload_pack.gsub(/.*git-upload-pack/, "").
-               lines.find { |l| l.include?(" HEAD") }
+        line = upload_pack.gsub(/.*git-upload-pack/, "")
```

Comment:
This is unrelated to my changes 😢

Reply:
Definitely not fault of these changes, I guess CodeQL checks changed lines, but if it finds any issues, it does not care whether the issues were present before the changes. I think it should, but it doesn't.

I did take a look anyways and provided a potential fix at https://github.com/dependabot/dependabot-core/pull/7792.

Reply:
I merged:
* #7792

Reply:
Yeah, I created a bunch of conflicts for this PR today, but I will stay quiet now until this lands. I recommend to regenerate the changes from scratch instead of fixing conflicts 😅.